src/rendering/webgl/programs/edge.clamped.ts

```diff
@@ -19,14 +19,29 @@ export default class EdgeClampedProgram extends EdgeRectangleProgram {
   getDefinition() {
     return {
       ...super.getDefinition(),
-      ARRAY_ITEMS_PER_VERTEX: 6,
       VERTEX_SHADER_SOURCE,
       ATTRIBUTES: [
-        { name: "a_position", size: 2, type: FLOAT },
+        { name: "a_positionStart", size: 2, type: FLOAT },
+        { name: "a_positionEnd", size: 2, type: FLOAT },
         { name: "a_normal", size: 2, type: FLOAT },
         { name: "a_color", size: 4, type: UNSIGNED_BYTE, normalized: true },
         { name: "a_radius", size: 1, type: FLOAT },
       ],
+      CONSTANT_ATTRIBUTES: [
+        // If 0, then position will be a_positionStart
+        // If 1, then position will be a_positionEnd
+        { name: "a_positionCoef", size: 1, type: FLOAT },
+        { name: "a_normalCoef", size: 1, type: FLOAT },
+        { name: "a_radiusCoef", size: 1, type: FLOAT },
+      ],
+      CONSTANT_DATA: [
+        [0, 1, 0],
+        [0, -1, 0],
```

Comment:
I wish prettier could let us align the data array to be beautiful :)

Reply:
agreed ^^